app/assets/javascripts/pageflow/editor/models/file_uploader.js

```diff
@@ -15,13 +17,31 @@ pageflow.FileUploader = pageflow.Object.extend({
       fileType: fileType
     });
 
+    var uploadNestedFile = editor.nextUploadTargetFile;
+
+    if (uploadNestedFile){
+      if (fileType.topLevelType) {
+        throw(new pageflow.TopLevelTypeError(upload));
+      }
+      file.set({parent_file_id: editor.nextUploadTargetFile.get('id'),
+                parent_file_model_type: editor.nextUploadTargetFile.fileType().typeName});
+    }
+    else if (!fileType.topLevelType) {
+      throw(new pageflow.NestedTypeError(upload));
+    }
+
```

Comment:
That is true. However, we have a validation in the Rails backend for that https://github.com/codevise/pageflow/pull/660/files#diff-647fc1f26572e8132240adef057a5ebcR22